routing/route/route.go

```diff
@@ -47,6 +50,15 @@ type Hop struct {
 	// and the last 2 bytes are the output index for the channel.
 	ChannelID uint64
 
+	// NextChanID is the unique channel ID that references the _outgoing_
+	// channel ID that follows this hop. This field follows the same
+	// semantics as the NextAddress field in the onion: it should be set to
+	// zero to indicate the terminal hop.
+	//
+	// NOTE: This field is only relevant when packing a hop payload with
+	// the PackHopPayload method.
+	NextChanID uint64
```

Comment:
How does this duplicate data? It's information that isn't present in the hop struct as is, it also adds additional contexts to the hop struct, and isn't serialized anywhere. If the pack method is extended, then you aren't able to pack a hop in isolation, as you need either the entire next hop, or to pass in this information manually.

Reply:
I don't think we're concerned about 8 bytes here and there at this moment. If we want to optimize struct size, then we can make a clean sweep to do proper struct alignment across the entire code base which will save us much more space w.r.t our in-memory representation. This doesn't appear to be a blocker, and in the future, if we want to reclaim those 8 bytes, we can modify this method as it's just an in-memory representation. 

If you look at the code, the current behavior is also symmetric for the legacy+modern format: set the next channel ID in the target struct, then encode the data.

Reply:
I already recognized in my comment above that efficiency probably doesn't matter in this case. 

Changing the representation later is not that simple, because code may be build on top of this that uses the new field. This is more work to reverse than just changing the struct.

I am no fan of duplicated data in a struct because it can be confusing and prone to bugs. A future dev writing a function that manipulates a route and forgets to update one or the other field. If you decide to go ahead with this change, then at least rename the existing field to `PrevChanID`.